de_workloads/data_processing/silver/data_factory/data.tf

```diff
@@ -0,0 +1,4 @@
+data "azurerm_data_factory" "factory" {
+  name                = var.data_factory
```

Comment:
where is this variable defined in  local directory?

Reply:
Just checking what you mean here. The pipeline picks this up from the ADO variable group and passes it to the `TF_VAR_data_factory`

Reply:
what I meant here is definition of this variable var.data_factory, with description and default it needs to be present in the same directory

Reply:
That's there in `vars.tf` in the same directory. It has a description, I don't think it should have a default as an explicit value needs to be passed each time.